include/zephyr/bluetooth/audio/mcc.h

```diff
@@ -67,6 +67,7 @@ typedef void (*bt_mcc_read_player_name_cb)(struct bt_conn *conn, int err, const
 typedef void (*bt_mcc_read_icon_obj_id_cb)(struct bt_conn *conn, int err, uint64_t id);
 #endif /* CONFIG_BT_OTS_CLIENT */
 
+#if defined(CONFIG_BT_MCC_READ_MEDIA_PLAYER_ICON_URL)
```

Comment:
There is no reason to guard the `typedef`s of the callbacks - Suggest to remove the guards for those.